### ISSUE TITLE
[CST] Incorporating claim letter on-ramps into main CST navigation

### ISSUE THREAD

**Issue by skylerschain** [department-of-veterans-affairs/va.gov-team#81241]
## Value Proposition

As a claimant checking claim status in CST, I want to have a way to find my claim letters when I have an open claim even if I don't have a specific alert related to those letters, so that I always can find claim letters I might need to download. 

## Background Context

See [original ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/79846) and [CAIA ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/81061) for more details. 

We want a way for Veterans to be able to navigate from the CST home page `va.gov/track-claims/your-claims` directly to the Claim Letters page `va.gov/track-claims/your-claim-letters` without having to go through an alert (like we use currently for decision letters). 

## Outcome, Success Measure, KPI(S), and Tracking Link
- Add here

## Design

[Spec in Figma](https://www.figma.com/file/BUdMIxAiIutG12rZxZ0cg1/Claim-Letters-Navigation-and-5103-alert?type=design&node-id=1451-13269&mode=design&t=p1rrPPiOMa7kxdl2-4)

Proposed: Add "Claim Letters" section and link to CST home page
![CSTHomePage](https://github.com/department-of-veterans-affairs/va.gov-team/assets/95102058/a88527c5-4068-491d-8134-968c6098d5b0)

### Enablement team (if needed)
@jstrothman 

## Engineering 
- Add here

## Out of scope
- Add here

## Open questions
- Are there any IA issues with this? 
- Is the "On this page" element still necessary? If so, we will need to update it to include this. If not, maybe we can remove it.

## Tasks
- [ ] _Add "Claim letters" header to CST home page below "Your claims, decision reviews, or appeals" section_
- [ ] _Task_

## Definition of Done
- [ ] Meets acceptance criteria
- [ ] Passed E2E testing (90% coverage)
- [ ] Passed unit testing (90% coverage)
- [ ] Passed integration testing (if applicable)
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Team approved production verification process
- [ ] Design performs design QA and verifies the implementation matches the design spec
- [ ] Accessibility specialist performs accessibility review (in code or design)
- [ ] Engineering identifies staging users required to test and shares account and credentials with design and product
- [ ] Product performs functional QA and verifies acceptance criteria was met

## Acceptance Criteria
- [ ] A claim letters link with a clear title description is added as an "on this page" navigation link.
- [ ] Below claim cards, there is a brief description of what users can do in the claim claim letters section with a link to the claim letters section of CST

**Comment by skylerschain** [comment]
Assigning to me until language and IA questions have been finalized.

**Comment by skylerschain** [comment]
Added link to final spec in **Design** section. This is ready for handoff.

**Comment by pmclaren19** [comment]
@skylerschain this is now in Staging/Prod. Please review for QA Design and let me know what text changes you are wanting.

**Comment by skylerschain** [comment]
Revised mock [here](https://www.figma.com/file/BUdMIxAiIutG12rZxZ0cg1/Claim-Letters-Navigation-and-5103-alert?type=design&node-id=1451-12998&mode=design&t=kK21zyOFacpi5Eh7-4) with language changes to remove "appeals" since we are pausing shipping board appeal letters.

Screenshot 

<img width="499" alt="Screenshot 2024-05-03 at 11 57 14 AM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/95102058/520f9115-13e3-4a0e-9031-1885fcea4e97">

**Comment by pmclaren19** [comment]
> Revised mock [here](https://www.figma.com/file/BUdMIxAiIutG12rZxZ0cg1/Claim-Letters-Navigation-and-5103-alert?type=design&node-id=1451-12998&mode=design&t=kK21zyOFacpi5Eh7-4) with language changes to remove "appeals" since we are pausing shipping board appeal letters.
> 
> Screenshot
> 
> <img alt="Screenshot 2024-05-03 at 11 57 14 AM" width="499" src="https://private-user-images.githubusercontent.com/95102058/327813429-520f9115-13e3-4a0e-9031-1885fcea4e97.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3NTU4MDUsIm5iZiI6MTcxNDc1NTUwNSwicGF0aCI6Ii85NTEwMjA1OC8zMjc4MTM0MjktNTIwZjkxMTUtMTNlMy00YTBlLTkwMzEtMTg4NWZjZWE0ZTk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAzVDE2NTgyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2NjQxNzJlYzgwMjM2ZTAwNmM2NmU2ZDM2MGJlM2M1NmYzYzMyY2YwNTliNzYzNmQ0ZmVhMGNlOWZjNTBiZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.eMLcAG8r4XOebOks5YJKlMjJVf0d_kAnL_2YdQD3zOU">

Talked to @skylerschain and we will be adding these changes...

- When the ff is disabled the subtext will read `You can download your decision letters online. You can also get other letters related to your claims.`
- When the ff is enabled the subtext will read  `You can download your decision letters online. You can also get other letters related to your claims and appeals.`

**Comment by skylerschain** [comment]
I tested this out too and if the user has no claim letters, the footer just
takes up a large portion of the screen and it kinda looks like the focus is
set to the footer (even though it isn't). Not sure that's what you're
experiencing, Julie.

**Comment by jstrothman** [comment]
> Would you mind giving me more details like what user are you on in Staging that your seeing this? Also might be worth doing a hard refresh and logging out and back in and trying again.

vets.gov.user+54@gmail.com

That user has a letter, but the page loads at the footer while waiting for letters to load (while the loading spinner displays, which takes a while)

**Comment by pmclaren19** [comment]
Merged update subtext and verified this is working correctly in staging.

**Comment by pmclaren19** [comment]
> > Would you mind giving me more details like what user are you on in Staging that your seeing this? Also might be worth doing a hard refresh and logging out and back in and trying again.
> 
> [vets.gov.user+54@gmail.com](mailto:vets.gov.user+54@gmail.com)
> 
> That user has a letter, but the page loads at the footer while waiting for letters to load (while the loading spinner displays, which takes a while)

Hey @jstrothman ,

Took a look at that user and this is what I was seeing:

https://github.com/department-of-veterans-affairs/va.gov-team/assets/141954992/d43e272c-8569-454f-974a-8223dde679a6

Is this what you are  seeing or are you seeing something different? I took a look at our logic and right now our logic says that if there the page is loading we dont set the focus, but once the page is no longer loading we set the page focus to the top of the page (where the breadcrumbs are). Please let me know if you have any concerns around this logic! Thanks

**Comment by davidakennedy** [comment]
After the team talked about this in standup this morning, I wanted to test it because your description @jstrothman [reminded me of a similar ticket I filed](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2124).

That ticket occurred in one of the 1010 forms, and normally happens when you have a long screen of text, then you activate a button to go to the next screen, which is much shorter. Very similar to this use case. I did some testing, and couldn't reproduce it here. Here's a GIF of what I found, similar to @pmclaren19. That doesn't mean the bug isn't happening, it may be hard to reproduce. I tried at several different screen widths. The element in the console at the bottom of the GIF shows what has active focus.

![Animated GIF of focus working properly and going to the breadcrumb menu](https://github.com/department-of-veterans-affairs/va.gov-team/assets/1473618/ddc95832-a608-45b5-8ac2-c7ab8ec80992)

**Comment by pmclaren19** [comment]
@davidakennedy , this looks to be matching the code... in the video you added, if you dont scroll to the top and instead wait for the page to load, when its done does it bring you to the top of the page for focus? Based on the code logic, it should bring you to the top of the page once the loading is complete but since you can recreate, want to verify this. 

Like I said we can change when the page focus occurs so if you want it to always happen, even if the page isnt loaded we could do that. Let m eknow @jstrothman

**Comment by jstrothman** [comment]
@pmclaren19 I'd like to avoid the page loading with the footer displayed as it feels like it's an error until you scroll up. I'm not sure why we wait for va-breadcrumbs.hydrated (or why that's the focus) -- @davidakennedy do you have a recommendation of what we should focus on that supports accessibility and also helps the user know they've landed where they intended?

**Comment by davidakennedy** [comment]
@pmclaren19

> if you dont scroll to the top and instead wait for the page to load, when its done does it bring you to the top of the page for focus?

It does! I scrolled up manually there to make sure it was still loading. But it worked like that for me when I tested each time.

@jstrothman @pmclaren19 Re: Where to send focus: In most of our forms we either send focus to a h3 that's unique to the page. If no h3 is there, we send it to the breadcrumbs. In this case, the two best options are:

- Sending it to the top of the page, like Peri mentions. This is a good option, although inconsistent with most other forms and apps. Peri, when you say "top of the page" do you mean the top of the html container?
- Sending focus to the h1: "Your VA claim and appeal letters". This feels best to me if it's feasible technically. It's more specific than the breadcrumbs and conveys what the page is about. The downside is that Veterans would need to navigate backward to get to the breadcrumbs, less of an issue in my mind. We usually don't send focus to the h1 on forms because it stays the same on all pages because of the h1/stepper issue. But that's not in play here.

**Comment by pmclaren19** [comment]
@davidakennedy we are currently using a util function that scrolls to the tops of the page and sets the page focus to the va-breadcrumbs if breadcrumbs exist and if they dont then it sets it to the h1. This is used in several other places throughout cst. @jstrothman ill create a pr to update the focus to occur regardless of if a page is still being loaded.

**Comment by pmclaren19** [comment]
Merged this change this morning and tested in Staging. It all looks good to me. @jstrothman please let me know if you have any questions/concerns!
